tests/php/test-class-amp-youtube-embed-handler.php

```diff
@@ -188,6 +190,93 @@ public function test__conversion( $source, $expected, $fallback_for_expected = n
 		}
 	}
 
+	/**
+	 * Gets the test data for test_get_video_id_from_url().
+	 *
+	 * @return array The test data.
+	 */
+	public function get_video_id_data() {
+		return [
+			'basic_url'                        => [
+				'https://www.youtube.com/watch?v=XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'short_url'                        => [
+				'https://youtu.be/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'url_with_underscore'              => [
+				'https://www.youtube.com/watch?v=CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'short_url_with_underscore'        => [
+				'https://youtu.be/CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'url_with_hyphen'                  => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8',
+				'xo68-iWaKv8',
+			],
+			'url_with_hyphen_and_query_string' => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8&w=800&h=400',
+				'xo68-iWaKv8',
+			],
+			'url_with_hyphen_and_query_string_dimensions_before_id' => [
+				'https://www.youtube.com/watch?w=800&h=400&v=xo68-iWaKv8',
+				'xo68-iWaKv8',
+			],
+			'embed_url'                        => [
+				'http://www.youtube.com/embed/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'embed_url_ending_in_query_param'  => [
+				'http://www.youtube.com/embed/XOY3ZUO6P0k?rel=0',
+				'XOY3ZUO6P0k',
+			],
+			'v_segment_url'                    => [
+				'http://youtube.com/v/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'e_segment_url'                    => [
+				'http://youtube.com/e/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'vi_segment_url'                   => [
+				'http://youtube.com/vi/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'vi_query_param_url'               => [
+				'http://youtube.com/?vi=XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'nocookie_url'                     => [
+				'//www.youtube-nocookie.com/embed/XOY3ZUO6P0k?rel=0',
+				'XOY3ZUO6P0k',
+			],
+			'false_because_no_id'              => [
+				'http://youtube.com/?wrong=XOY3ZUO6P0k',
+				false,
+			],
+		];
```

Comment:
Other cases to test that I think would currently break:
```
https://www.youtube.com/account
https://www.youtube.com/playlist?list=WL
```

Reply:
Good point, these should return `false`, but currently return `account` and `playlist`